src/test/java/org/mockitousage/matchers/CapturingArgumentsTest.java

```diff
@@ -314,4 +330,53 @@ public void captures_correctly_when_captor_used_on_pure_vararg_method() throws E
         verify(mock).varargs(eq(42), argumentCaptor.capture());
         Assertions.assertThat(argumentCaptor.getValue()).contains("capturedValue");
     }
+
+    @Test
+    public void should_capture_only_list_instances() throws Exception {
+        //given
+        ArgumentCaptor<List> captor = ArgumentCaptor.forClass(List.class);
+
+        //when
+        mock.forCollection(new ArrayList<String>());
+        mock.forCollection(new HashSet<String>());
+        mock.forCollection(new LinkedList<String>());
+
+        //then
+        verify(mock, times(2)).forCollection(captor.captureIf(new InstanceOf(List.class)));
+        assertThat(2, is(equalTo(captor.getAllValues().size())));
+    }
+
+    @Test
+    public void should_capture_when_value_is_equal_to_a_certain_value() throws Exception {
+        //given
+        ArgumentCaptor<String> captor = ArgumentCaptor.forClass(String.class);
+
+        //when
+        mock.simpleMethod(FOO, 1);
+        mock.simpleMethod(BAR, 2);
+
+        //then
+        verify(mock).simpleMethod(captor.captureIf(new Equals(FOO)), anyInt());
+        assertThat(1, is(equalTo(captor.getAllValues().size())));
+        assertThat(FOO, is(equalTo(captor.getValue())));
+    }
+
+    @Test
+    public void should_capture_when_value_is_less_then() {
+        //given
+        ArgumentCaptor<Integer> captor = ArgumentCaptor.forClass(Integer.class);
+
+        //when
+        mock.simpleMethod(FOO, 1);
+        mock.simpleMethod(BAR, 2);
+        mock.simpleMethod(FOO, 3);
+        mock.simpleMethod(BAR, 4);
+        
+        mock.simpleMethod(FOO, 5);
+
+        //then
+        verify(mock, times(2)).simpleMethod(anyString(), captor.captureIf(new LessThan<Integer>(3)));
```

Comment:
Hi Christian,
Thanks for the quick response. I check the source code of AdditionalMatchers. With the current implementation I can't use it, because I would need the ArgumentMatcher used in the is, lt, etc methods. For ex. AdditionMatcher.is will return 0 all the time. Somehow I would need the evaluation result of the argument matcher used behind.